### ISSUE TITLE
Peer seed from cdn failed, but task status is success

### ISSUE THREAD

**Issue by gaius-qi** [dragonflyoss/Dragonfly2#968]
### Bug report:

Actions: https://github.com/dragonflyoss/Dragonfly2/actions/runs/1637412768

```text
core.log:{"level":"info","ts":"2021-12-30 12:23:08.984","caller":"rpcserver/rpcserver.go:59","msg":"register peer task, req: &scheduler.PeerTaskRequest{state:impl.MessageState{NoUnkeyedLiterals:pragma.NoUnkeyedLiterals{}, DoNotCompare:pragma.DoNotCompare{}, DoNotCopy:pragma.DoNotCopy{}, atomicMessageInfo:(*impl.MessageInfo)(0xc00033f400)}, sizeCache:0, unknownFields:[]uint8(nil), Url:\"http://file-server.dragonfly-e2e.svc/kind/etc/hostname\", UrlMeta:(*base.UrlMeta)(0xc0003282a0), PeerId:\"10.244.0.5-1-380f469c-f2b5-4245-bd5c-1637639aa4e9\", PeerHost:(*scheduler.PeerHost)(0xc000355ae0), HostLoad:(*base.HostLoad)(nil), IsMigrating:false}","taskId":"33e4d783617fa2417ef76782d54c4206646c71a23ee7b4c94e885048e6ff19b5","peerID":"10.244.0.5-1-380f469c-f2b5-4245-bd5c-1637639aa4e9"}
core.log:{"level":"info","ts":"2021-12-30 12:23:08.985","caller":"core/service.go:266","msg":"add new task 33e4d783617fa2417ef76782d54c4206646c71a23ee7b4c94e885048e6ff19b5","taskId":"33e4d783617fa2417ef76782d54c4206646c71a23ee7b4c94e885048e6ff19b5"}
core.log:{"level":"info","ts":"2021-12-30 12:23:08.985","caller":"rpcserver/rpcserver.go:145","msg":"task is unsuccessful and needs to be register","taskId":"33e4d783617fa2417ef76782d54c4206646c71a23ee7b4c94e885048e6ff19b5","peerID":"10.244.0.5-1-380f469c-f2b5-4245-bd5c-1637639aa4e9"}
core.log:{"level":"info","ts":"2021-12-30 12:23:08.985","caller":"supervisor/cdn.go:84","msg":"start seed task 33e4d783617fa2417ef76782d54c4206646c71a23ee7b4c94e885048e6ff19b5"}
core.log:{"level":"warn","ts":"2021-12-30 12:23:08.986","caller":"core/events.go:144","msg":"startReportPieceResultEvent: no parent node is currently available，reschedule it later","taskId":"33e4d783617fa2417ef76782d54c4206646c71a23ee7b4c94e885048e6ff19b5","peerID":"10.244.0.5-1-380f469c-f2b5-4245-bd5c-1637639aa4e9","stacktrace":"d7y.io/dragonfly/v2/scheduler/core.startReportPieceResultEvent.apply\n\t/go/src/d7y.io/dragonfly/v2/scheduler/core/events.go:144\nd7y.io/dragonfly/v2/scheduler/core.(*baseWorker).start\n\t/go/src/d7y.io/dragonfly/v2/scheduler/core/worker.go:83"}
core.log:{"level":"info","ts":"2021-12-30 12:23:08.989","caller":"supervisor/cdn.go:159","msg":"task 33e4d783617fa2417ef76782d54c4206646c71a23ee7b4c94e885048e6ff19b5 add piece peer_id:\"10.244.0.21-9-3c91db12-2afd-4fba-b320-5b077a2d14b4_CDN\"  host_uuid:\"dragonfly-cdn-1.cdn.dragonfly-system.svc.cluster.local-8003_CDN\"  piece_info:{range_size:19  piece_md5:\"afcda301db1b5299cf3d7fec3d964ec9\"  piece_style:1}"}
core.log:{"level":"info","ts":"2021-12-30 12:23:08.989","caller":"supervisor/cdn.go:86","msg":"finish seed task 33e4d783617fa2417ef76782d54c4206646c71a23ee7b4c94e885048e6ff19b5, task status is Running"}
core.log:{"level":"error","ts":"2021-12-30 12:23:08.989","caller":"core/service.go:294","msg":"seed task failed: cannot find host dragonfly-cdn-1.cdn.dragonfly-system.svc.cluster.local-8003_CDN: init cdn peer failed","taskId":"33e4d783617fa2417ef76782d54c4206646c71a23ee7b4c94e885048e6ff19b5","stacktrace":"d7y.io/dragonfly/v2/scheduler/core.(*SchedulerService).GetOrAddTask.func1\n\t/go/src/d7y.io/dragonfly/v2/scheduler/core/service.go:294"}
core.log:{"level":"info","ts":"2021-12-30 12:23:08.992","caller":"core/events.go:128","msg":"startReportPieceResultEvent: no need schedule parent because peer is back source peer","taskId":"33e4d783617fa2417ef76782d54c4206646c71a23ee7b4c94e885048e6ff19b5","peerID":"10.244.0.5-1-380f469c-f2b5-4245-bd5c-1637639aa4e9"}
core.log:{"level":"info","ts":"2021-12-30 12:23:14.941","caller":"rpcserver/rpcserver.go:59","msg":"register peer task, req: &scheduler.PeerTaskRequest{state:impl.MessageState{NoUnkeyedLiterals:pragma.NoUnkeyedLiterals{}, DoNotCompare:pragma.DoNotCompare{}, DoNotCopy:pragma.DoNotCopy{}, atomicMessageInfo:(*impl.MessageInfo)(0xc00033f400)}, sizeCache:0, unknownFields:[]uint8(nil), Url:\"http://file-server.dragonfly-e2e.svc/kind/etc/hostname\", UrlMeta:(*base.UrlMeta)(0xc0003c6690), PeerId:\"10.244.0.28-1-886c8db1-e8f3-4f51-94b7-18e0d4f5c024\", PeerHost:(*scheduler.PeerHost)(0xc00051c000), HostLoad:(*base.HostLoad)(nil), IsMigrating:false}","taskId":"33e4d783617fa2417ef76782d54c4206646c71a23ee7b4c94e885048e6ff19b5","peerID":"10.244.0.28-1-886c8db1-e8f3-4f51-94b7-18e0d4f5c024"}
core.log:{"level":"info","ts":"2021-12-30 12:23:14.941","caller":"rpcserver/rpcserver.go:78","msg":"task has been successful","taskId":"33e4d783617fa2417ef76782d54c4206646c71a23ee7b4c94e885048e6ff19b5","peerID":"10.244.0.28-1-886c8db1-e8f3-4f51-94b7-18e0d4f5c024"}
core.log:{"level":"info","ts":"2021-12-30 12:23:14.941","caller":"rpcserver/rpcserver.go:83","msg":"task size scope is tiny and return piece content directly","taskId":"33e4d783617fa2417ef76782d54c4206646c71a23ee7b4c94e885048e6ff19b5","peerID":"10.244.0.28-1-886c8db1-e8f3-4f51-94b7-18e0d4f5c024"}
```

### Environment:
- Dragonfly version:
- OS:
- Kernel (e.g. `uname -a`):
- Others:

**Comment by jim3ma** [comment]
Logs: [cdn-compatibility-e2e-tests-logs.zip](https://github.com/dragonflyoss/Dragonfly2/files/7793088/cdn-compatibility-e2e-tests-logs.zip)

**Comment by jim3ma** [comment]
Fixed in #970
